geomstats/vectorization.py

```diff
@@ -106,6 +111,36 @@ def repeat_point(point, n_reps=2, expand=False):
     return gs.repeat(gs.expand_dims(point, 0), n_reps, axis=0)
 
 
+def _is_not_none(value):
+    return value is not None
+
+
+def repeat_out(space, out, *points, out_shape=()):
+    """Repeat out shape after finding batch shape.
+
+    Parameters
+    ----------
+    space : Manifold
+        Space to which point belongs.
+    out : array-like
+        Output to be repeated
+    point : array-like
+        Point belonging to the space.
+    out_shape : tuple
+        Indicates out shape for no batch computations.
+
+    Returns
+    -------
+    out : array-like
+        If no batch, then input is returned. Otherwise it is broadcasted.
+    """
+    points = filter(_is_not_none, points)
```

Comment:
I have trouble reading through this because some private functions do not have docstrings (as is conventional) and the public functions have short docstrings.

Any chance you could add details + 1-2 examples in the docstrings, even if private ones, to explain the logic?

Reply:
I was also confused in the _get_max_ndim and the line point_max_ndim = point[0]

--> why is the ndim an element of point? 

docstrings would help for this.

Reply:
I'll improve the docstrings.

The main idea is that I want to have these methods working for the cases where `point` is `None`, because in a lot of our methods we have something like `base_point=None` and then I want to use something like `repeat_out(self.space, out, base_point)` without having to check if there's `None` (it simplifies a lot the vectorization logic there).

therefore, before checking `batch_shape`, I remove the `None` from the list of points (that's the role of `filter`).